packages/yourloops/pages/signup/signup-profile-form.tsx

```diff
@@ -70,38 +70,38 @@ const SignUpProfileForm: FunctionComponent<SignUpFormProps> = (props) => {
 
   const isFormEmpty = useMemo<boolean>(() => {
     return !_.some(errors) &&
-      !state.profileFirstname &&
-      !state.profileLastname &&
-      !state.profileCountry &&
-      !state.hcpProfession
-  }, [errors, state])
+      !signupForm.profileFirstname &&
+      !signupForm.profileLastname &&
+      !signupForm.profileCountry &&
+      !signupForm.hcpProfession
+  }, [errors, signupForm])
 
   const onChange = (value: string | unknown, key: SignupFormKey): void => {
-    dispatch({ type: 'EDIT_FORMVALUE', key, value: value as string })
+    setSignupForm(prevState => ({ ...prevState, [key]: value }))
```

Comment:
This bit of code could have been in the context directly (it would have avoided duplication)